Sources/Sugar/Authentication/UserType.swift

```diff
@@ -13,7 +13,25 @@ public protocol UserType {
     func update(with: Update) throws
 
     static func logIn(with: Login, on worker: DatabaseConnectable) -> Future<Self>
-    static func preRegister(with: Registration, on worker: DatabaseConnectable) throws -> Future<Void>
+    /// Performs work, such as validation, before the user is being created.
+    ///
+    /// - Parameters:
+    ///   - with: The registration object.
```

Comment:
This reads a bit weird. I ran into this as well when documenting protocol methods. If you have a label that is not the variable name the autogenerated docs use the label. If you do `with registration: Registration,` the API docs will have `registration` instead of `with`.